pubsub/jetstream/jetstream.go

```diff
@@ -88,8 +96,26 @@ func (js *jetstreamPubSub) Features() []pubsub.Feature {
 }
 
 func (js *jetstreamPubSub) Publish(req *pubsub.PublishRequest) error {
-	js.l.Debugf("Publishing topic %v with data: %v", req.Topic, req.Data)
-	_, err := js.jsc.Publish(req.Topic, req.Data)
+	var opts []nats.PubOpt
+	var msgID string
+
+	event, err := unmarshallCloudEvent(req)
```

Comment:
Yeah, it felt very heavy (having to add another dependency). I think this should be provided by some component utils so all components can make use of it.

I did see this function that looks like it could be useful, but it requires too many arguments to be instantly useful:

https://github.com/dapr/components-contrib/blob/0326f139c5a7287c23a6c15c998f48f63ef34b4d/pubsub/envelope.go#L128-L131

How would you suggest to do it?

Reply:
Please do not modify anything in this shared library. Changes to the envelope or our handling of cloud events really needs a separate proposal and issue.

Reply:
This is something that would need to happen in the Dapr/Dapr repo first, but perhaps the PublishRequest could be updated to have an ID field. Here is the proto:
https://github.com/dapr/dapr/blob/2ca43ec3c91b0c1043feb6f1872cdaf4aa3d8149/dapr/proto/runtime/v1/dapr.proto#L269

Reply:
@berndverst 

> perhaps the PublishRequest could be updated to have an ID field

This would be great, would make things a lot easier for all of the components implementing deduplication.

In the meantime, I wonder if I drop the extra dependency and use `FromCloudEvent` as also mentioned in your comment: https://github.com/dapr/components-contrib/issues/2200#issuecomment-1284737198

Reply:
@berndverst, I've tried to use `FromCloudEvent`, but I'm a little stuck.

For one, I have to provude superfluous arguments when only the request data is necessary. Also, because it produces a `map[string]interface{}`, I still have to get `event["id"]` and cast it as a string, and error check it.

So using the official cloudevents package does seems a lot safer and convenient.

Would you be ok to accept as is?

Reply:
Update: I'd forgotten how to type cast in Go. I've made a revision that removes cloudevents package and uses `FromCloudEvent` with some checks.